src/transport/multicast/link/task/read.c

```diff
@@ -91,14 +91,18 @@ void *_zp_multicast_read_task(void *ztm_arg) {
             ret = _z_transport_message_decode_na(&t_msg, &zbuf);
             if (ret == _Z_RES_OK) {
                 ret = _z_multicast_handle_transport_message(ztm, &t_msg, &addr);
-
+                bool should_close = (_Z_MID(t_msg._header) == _Z_MID_CLOSE);
```

Comment:
The read task in multicast should never be stopped in response of a close message.
The same read task is used to read all multicast messages from all the peers, so one peer could stop the communication with all the other peers.

Hence, proposed changes to multicast should be reverted.